test/integration/graph/test_require_same_pkg_versions.py

```diff
@@ -186,3 +186,190 @@ def build(self):
+            def package(self):
+                echo = f"@echo off\necho MYGCC={self.version}!!"
+                save(self, os.path.join(self.package_folder, "bin", "mygcc.bat"), echo)
+                save(self, os.path.join(self.package_folder, "bin", "mygcc.sh"), echo)
+                os.chmod(os.path.join(self.package_folder, "bin", "mygcc.sh"), 0o777)
+            """)
+
+    project = textwrap.dedent(r"""
+       import os, platform
+       from conan import ConanFile
+       from conan.tools.files import save, chdir
+       class Pkg(ConanFile):
+           name = "project"
+           version = "1.0"
+           def build_requirements(self):
+               self.tool_requires("wrappera/1.0")
+               self.tool_requires("wrapperb/1.0")
+           def build(self):
+               ext = "bat" if platform.system() == "Windows" else "sh"
+               self.run(f'mygcc.{ext}')
+       """)
+    c.save({"gcc/conanfile.py": gcc,
+            "wrappera/conanfile.py": GenConanfile("wrappera", "1.0").with_requires("gcc/1.0"),
+            "wrapperb/conanfile.py": GenConanfile("wrapperb", "1.0").with_requires("gcc/1.0"),
+            "project/conanfile.py": project})
+
+    c.run("create gcc --version=1.0")
+    c.run("create wrappera")
+    c.run("create wrapperb")
+
+    c.run("build project")
+    assert "MYGCC=1.0!!" in c.out
+
+
+def test_require_different_versions_transitive():
+    """ Similar as above, but with transitive
+    """
+    c = TestClient()
+    gcc = textwrap.dedent(r"""
+        import os
+        from conan import ConanFile
+        from conan.tools.files import save
+        class Pkg(ConanFile):
+            name = "gcc"
+            package_type = "application"
+            def package(self):
+                echo = f"@echo off\necho MYGCC={self.version}!!"
+                save(self, os.path.join(self.package_folder, "bin", "mygcc.bat"), echo)
+                save(self, os.path.join(self.package_folder, "bin", "mygcc.sh"), echo)
+                os.chmod(os.path.join(self.package_folder, "bin", "mygcc.sh"), 0o777)
+            """)
+
+    project = textwrap.dedent("""
+        from conan import ConanFile
+        class Pkg(ConanFile):
+            name = "project"
+            version = "1.0"
+            def build_requirements(self):
+                self.tool_requires("wrappera/1.0")
+                self.tool_requires("wrapperb/1.0")
+            """)
+    c.save({"gcc/conanfile.py": gcc,
+            "wrappera/conanfile.py": GenConanfile("wrappera", "1.0").with_requires("gcc/1.0"),
+            "wrapperb/conanfile.py": GenConanfile("wrapperb", "1.0").with_requires("gcc/2.0"),
+            "project/conanfile.py": project})
+
+    c.run("create gcc --version=1.0")
+    c.run("create gcc --version=2.0")
+    c.run("create wrappera")
+    c.run("create wrapperb")
+    c.run("install project", assert_error=True)
+    assert "ERROR: Runtime Error: Could not process 'wrapperb/1.0' with 'gcc/2.0'" in c.out
```

Comment:
Isn't this error message a bit vague as to what went wrong? Or does the full output make it more clear?

Reply:
Nop, it is mostly the full message. Certainly it can be improved to something more explicit and clear.